toolchain/check/member_access.cpp

```diff
@@ -87,20 +91,90 @@ static auto IsInstanceMethod(const SemIR::File& sem_ir,
   return false;
 }
 
+static auto LookupMemberNameInScope(Context& context,
+                                    Parse::MemberAccessExprId node_id,
+                                    SemIR::InstId /*base_id*/,
+                                    SemIR::NameId name_id,
+                                    SemIR::NameScopeId name_scope_id)
+    -> SemIR::InstId {
+  auto inst_id = name_scope_id.is_valid() ? context.LookupQualifiedName(
+                                                node_id, name_id, name_scope_id)
+                                          : SemIR::InstId::BuiltinError;
+  auto inst = context.insts().Get(inst_id);
+  // TODO: Track that this instruction was named within `base_id`.
```

Comment:
I may be missing something here... Isn't this just a matter of adding a tombstone to name_scope_id if LookupQualifiedName failed? Why is base_id necessary?

Is this even something that should be done here, versus as part of LookupQualifiedName?

Note, part of my thinking is that if `base_id` is an incomplete type and `name_scope_id` is invalid, the lookup will always be an error, so not sure how important it is to track that the field was named. (it will be diagnosed, and probably should not create a conflict with a later definition)

Reply:
This TODO isn't about that at all :)

Currently, for `A.B` we create a `name_ref` that refers to `B`, but doesn't mention that we looked for it in `A` at all.  The point of the TODO is that the `name_ref` instruction created on the next line should track the `base_id`.

I've changed the TODO to try to make this clearer (but this is pre-existing, just moved).